internal/webhook/hyperconverged/handler.go

```diff
@@ -154,64 +161,96 @@ func (h *handler) Handle(ctx context.Context, req admission.Request) admission.R
-	log.Info("list of nodes available for pod", "pod", pod.Name, "nodes", nodeNames)
+func (h *handler) patchPodWithNodeAffinity(pod *corev1.Pod, nodeNames []string, failoverMode string) admission.Response {
+	log.Info("list of nodes available for pod", "pod", pod.Name, "nodes", nodeNames, "failoverMode", failoverMode)
 
-	// add the node affinity to worker label name to the pod
+	// add the node affinity to the pod based on failover mode
 	newPod := pod.DeepCopy()
 	affinity := corev1.Affinity{}
 	nodeAffinity := corev1.NodeAffinity{}
-	preferredSchedulingTerm := []corev1.PreferredSchedulingTerm{}
-	schedulingTerm := corev1.PreferredSchedulingTerm{
-		Weight: 100,
-		Preference: corev1.NodeSelectorTerm{
-			MatchExpressions: []corev1.NodeSelectorRequirement{
-				{
-					Key:      KubernetesNodeHostNameLabel,
-					Operator: corev1.NodeSelectorOpIn,
-					Values:   nodeNames,
-				},
-			},
-		},
-	}
-	preferredSchedulingTerm = append(preferredSchedulingTerm, schedulingTerm)
 
-	shuffleObjects(*h.rng, preferredSchedulingTerm)
+	// Create node selector requirement
+	nodeSelectorRequirement := corev1.NodeSelectorRequirement{
+		Key:      KubernetesNodeHostNameLabel,
+		Operator: corev1.NodeSelectorOpIn,
+		Values:   nodeNames,
+	}
 
 	if newPod.Spec.Affinity == nil {
 		newPod.Spec.Affinity = &affinity
 	}
 	if newPod.Spec.Affinity.NodeAffinity == nil {
 		newPod.Spec.Affinity.NodeAffinity = &nodeAffinity
 	}
-	if newPod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution == nil {
-		newPod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = preferredSchedulingTerm
-	} else {
-		// add the preferred scheduling term to the existing preferred scheduling terms
-		newPod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = append(newPod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution, preferredSchedulingTerm...)
+
+	// Apply affinity based on failover mode
+	switch failoverMode {
+	case FailoverModeDurability:
+		// Required affinity - pod must be scheduled on one of these nodes
+		log.Info("applying required node affinity for durability mode", "pod", pod.Name)
+		requiredTerm := corev1.NodeSelectorTerm{
+			MatchExpressions: []corev1.NodeSelectorRequirement{nodeSelectorRequirement},
+		}
+		if newPod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+			newPod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{requiredTerm},
+			}
+		} else {
+			// Add to existing required terms
+			newPod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = append(
+				newPod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms,
+				requiredTerm,
+			)
+		}
+	case FailoverModeAvailability:
+		fallthrough
```

Comment:
suggestion: It could be nice to have an error or event if the mode specified was invalid?